CCS/3. Tutorials/7. 開啟 TensorFlow 自動混合精度運算與執行效能分析.md

```diff
@@ -1 +1,305 @@
+              optimizer=opt,
+              metrics=['accuracy'])
+elapsed_time = time.time() - start_time
+print('Time for model.compile:', elapsed_time)
+```
+
+:::info
+:bulb: The `KerasMNIST.py` (AMP is not opened) and `kerasMNIST-AMP.py` (AMP is opened)  program file can be downloaded and reference [here](https://github.com/TW-NCHC/AI-Services/tree/master/Tutorial_One).
+:::
+
+## Benchmark performance analysis:ResNet-50 v1.5
+
+The following table is based on five different settings to train the ResNet-50 v1.5 image recognition model and compare the computing performance between each other:
+
+| Group| AMP | XLA |  Precesion |Batch Size |
+| -------- | -------- | -------- | -------- | -------- |
+| 1| :negative_squared_cross_mark:     | :negative_squared_cross_mark:     | Double precision     | 128 (Baseline)     |
+| 2| :negative_squared_cross_mark:     |:negative_squared_cross_mark:     | Double precision     | 256     |
+| 3| :white_check_mark:     | :negative_squared_cross_mark:     | Mixed precision (single and half precision)    | 256     |
+| 4| :white_check_mark:     | :white_check_mark:     | Mixed precision (single and half precision)     | 256    |
+| 5| :white_check_mark:     | :white_check_mark:     | Mixed precision (single and half precision)   | 512    |
+
+:::info
+Note: Accelerated Linear Algebra (XLA) can optimize TensorFlow computing and accelerate execution speed.
+:::
+
+After create a TWCC Interactive Container (TensorFlow 19.08-p3:latest), the sample program ResNet-50 v1.5 already exists in the `/workspace/nvidia-examples/resnet50v1.5` directory. Create a new directory (example: results) under this directory to store the model data. Enter the command in the command line as follows:
+
+```bash=
+$ cd /workspace/nvidia-examples/resnet50v1.5
+$ mkdir results
+```
+
+###  Group 1. Treatment and control groups Baseline
+
+> Unactivate AMP | Double precision｜Batch size = 128
+
+The environment enables automatic mixed precision operation by default, so the execution control group must stop the automatic mixed precision operation in the environment variable in advance. To do so, directly enter the execution command in the command line as follows:
+
+```bash=
+$ export TF_ENABLE_AUTO_MIXED_PRECISION=0
+$ export TF_ENABLE_AUTO_MIXED_PRECISION_GRAPH_REWRITE=0
+$ python ./main.py --mode=training_benchmark --warmup_steps 200 \
+    --num_iter 500 --batch_size 128 --iter_unit batch --results_dir results \
+    --nouse_tf_amp  --nouse_xla
+```
+
+Using the following command to dynamically observe the usage of GPU memory every 10 seconds, and the usage of GPU memory that has been observed is about 25.43 GB. The GPU memory allocation capacity (32.48 GB) has not been used up:
+
+```bash=
+$ nvidia-smi --loop=10
+``` 
+
+![](https://cos.twcc.ai/SYS-MANUAL/uploads/upload_cf31d97e7819afb3bcdc7f8137c2ae5b.png)
+
+
+
+After the model training is completed, and it can process about 393.49 images per second:
```

Comment:
[SUG]

The model training completed, and It could process about 393.49 images per second: